### PR TITLE
Review spatial module test coverage

### PR DESCRIPTION
**Pull request by csmangum**
Add comprehensive test suites to the spatial module to improve test coverage from 79% to 97%.

This PR introduces new test files for `SpatialHashGrid`, `Quadtree`, `DirtyRegionTracker`, and advanced `SpatialIndex` features, addressing significant gaps in coverage and ensuring the reliability of core spatial functionalities like queries, updates, and flush policies.

---
<a href="https://cursor.com/background-agent?bcId=bc-809e8b9c-70bc-4bf0-9611-7b5d034b3e8b"><picture><source media="(prefers-color-scheme: dark)" srcset="https://cursor.com/open-in-cursor-dark.svg"><source media="(prefers-color-scheme: light)" srcset="https://cursor.com/open-in-cursor-light.svg">Open in Cursor</picture></a>&nbsp;<a href="https://cursor.com/agents?id=bc-809e8b9c-70bc-4bf0-9611-7b5d034b3e8b"><picture><source media="(prefers-color-scheme: dark)" srcset="https://cursor.com/open-in-web-dark.svg"><source media="(prefers-color-scheme: light)" srcset="https://cursor.com/open-in-web-light.svg">Open in Web</picture></a>

---

> [!NOTE]
> Adds extensive spatial module tests and a coverage report, boosting overall coverage to 97% with 300+ passing tests.
> 
> - **Tests (new)**:
>   - `tests/spatial/test_hash_grid.py`: Broad coverage of `SpatialHashGrid` (init, insert/remove/move, radius/range, nearest, edge cases, perf).
>   - `tests/spatial/test_quadtree.py`: Comprehensive `Quadtree`/`QuadtreeNode` (subdivision, queries, clear, stats, helpers, edge cases).
>   - `tests/spatial/test_dirty_regions.py`: Full `DirtyRegionTracker` scenarios (max_count, cleanup, timestamps, priorities, stats, batch ops).
>   - `tests/spatial/test_spatial_index_advanced.py`: Advanced `SpatialIndex` (flush policies, stale reads, batch enable/disable, priorities, flush methods, named indices with data_getter, nearest, range/nearby validation, auto cell size, quadtree stats, helpers, last flush time, defaults, edge cases).
> - **Tests (enhanced)**:
>   - `tests/spatial/test_batch_spatial_updates.py`: Expanded partial flushing/batch updates coverage.
> - **Docs/Reports**:
>   - `SPATIAL_TEST_COVERAGE_REPORT.md`: Coverage details (module breakdown, missing lines, execution) showing increase from 79% to 97%.
> - **Results**:
>   - All tests pass (302 total).
> 
> <sup>Written by [Cursor Bugbot](https://cursor.com/dashboard?tab=bugbot) for commit 140f650e3a6b223322021cb90fbb6b2e8995e393. This will update automatically on new commits. Configure [here](https://cursor.com/dashboard?tab=bugbot).</sup>

### CODE REVIEW

**Comment by csmangum** [comment]
bugbot run

File: tests/spatial/test_spatial_index_advanced.py [hunk 1]
```diff
@@ -0,0 +1,1433 @@
...
+        )
+        
+        entity = MockEntity("e1")
+        spatial_index.add_position_update(entity, (10.0, 10.0), (20.0, 20.0))
+        
+        # Remove from pending to set it up manually
+        spatial_index._pending_position_updates.clear()
+        spatial_index._pending_position_updates.append(
+            (entity, (10.0, 10.0), (20.0, 20.0), "agent", PRIORITY_NORMAL)
+        )
```
[the remark this answers is not in the document]
Comment (csmangum) [reply tests/spatial/test_spatial_index_advanced.py:1076]: @cursor resolve this

**Comment by csmangum** [comment]
@cursor eval these changes from the perspective of a senior engineer
